app/upgrades.go

```diff
@@ -0,0 +1,62 @@
+package app
+
+import (
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/cosmos/cosmos-sdk/x/nft"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	bech32ibckeeper "github.com/osmosis-labs/bech32-ibc/x/bech32ibc/keeper"
+	bech32ibctypes "github.com/osmosis-labs/bech32-ibc/x/bech32ibc/types"
+	leveragetypes "github.com/umee-network/umee/v2/x/leverage/types"
+	oracletypes "github.com/umee-network/umee/v2/x/oracle/types"
+)
+
+const UpgradeV3_0Plan = "v1.0-v3.0"
+
+func (app UmeeApp) RegisterUpgradeHandlers() {
+	// v3 upgrade handler performs upgrade from v1->v3
+	app.UpgradeKeeper.SetUpgradeHandler(
+		UpgradeV3_0Plan,
+		func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			ctx.Logger().Info("Upgrade handler execution", "name", UpgradeV3_0Plan)
+			err := setupBech32ibcKeeper(&app.bech32IbcKeeper, ctx)
+			if err != nil {
+				return nil, sdkerrors.Wrap(err, "Calypso Upgrade: Unable to upgrade, bech32ibc module not initialized")
```

Comment:
the upgrade plan is "v1.0-v3.0", but we still are going to call Calypso as well?

Reply:
yes, but for versioning wanted to use something more systematic.